Build failure: pkgsCross.mingwW64.gnugrep and pkgsCross.mingwW64.gnugrep

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix build .#pkgsCross.mingwW64.gnugrep`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/2v88s0kaqphqzr2djs8q0bxvm7qdkv8g-gnugrep-x86_64-w64-mingw32-3.12.drv' failed with exit code 2;
       last 25 log lines:
       >   CC       libgreputils_a-version-etc-fsf.o
       >   CC       libgreputils_a-wcrtomb.o
       >   CC       libgreputils_a-wctob.o
       >   CC       libgreputils_a-wctype.o
       >   CC       libgreputils_a-wctype-h.o
       >   CC       libgreputils_a-windows-mutex.o
       >   CC       libgreputils_a-windows-once.o
       >   CC       libgreputils_a-windows-recmutex.o
       >   CC       libgreputils_a-windows-rwlock.o
       >   CC       libgreputils_a-xmalloc.o
       >   CC       libgreputils_a-xalloc-die.o
       >   CC       libgreputils_a-xbinary-io.o
       > xbinary-io.c: In function 'xset_binary_mode_error':
       > xbinary-io.c:35:3: error: format not a string literal and no format arguments []
       >    35 |   error (exit_failure, errno,
       >       |   ^~~~~
       > cc1: some warnings being treated as errors
       > make[3]: *** [Makefile:6236: libgreputils_a-xbinary-io.o] Error 1
       > make[3]: *** Waiting for unfinished jobs....
       > make[3]: Leaving directory '/build/grep-3.12/lib'
       > make[2]: *** [Makefile:3310: all] Error 2
       > make[2]: Leaving directory '/build/grep-3.12/lib'
       > make[1]: *** [Makefile:2483: all-recursive] Error 1
       > make[1]: Leaving directory '/build/grep-3.12'
       > make: *** [Makefile:2424: all] Error 2
       For full logs, run:
         nix log /nix/store/2v88s0kaqphqzr2djs8q0bxvm7qdkv8g-gnugrep-x86_64-w64-mingw32-3.12.drv
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.34, NixOS, 25.11 (Xantusia), 25.11.20250621.4206c4c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/si3k775n94ax09w3cjgc5hr2vjs84v7m-source`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/text/gnugrep/default.nix
+++ b/pkgs/tools/text/gnugrep/default.nix
@@ -84,6 +84,10 @@ stdenv.mkDerivation {
     chmod +x $out/bin/egrep $out/bin/fgrep
   '';
 
+  env = lib.optionalAttrs stdenv.hostPlatform.isMinGW {
+    NIX_CFLAGS_COMPILE = "-Wno-error=format-security";
+  };
+
   meta = with lib; {
     homepage = "https://www.gnu.org/software/grep/";
     description = "GNU implementation of the Unix grep command";

